File: backend/src/models/Nota.js
'use strict';
const { Model, DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  class Nota extends Model {
    static associate(models) {
      // Associação de Nota com Cliente (muitos para um)
      Nota.belongsTo(models.Cliente, {
        foreignKey: 'cliente_id', // Usando `cliente_id` como chave estrangeira
        as: 'cliente'
      });
      
      // Associação de Nota com User (muitos para um)
      Nota.belongsTo(models.User, {
        foreignKey: 'criado_por_id', // Usando `criado_por_id` como chave estrangeira
        as: 'criador'
      });
    }
  }

  Nota.init({
    cliente_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'clientes',
        key: 'id'
      }
    },
    processo_id: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    texto: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    criado_por_id: {
      type: DataTypes.INTEGER,
      allowNull: true,
      references: {
        model: 'users',
        key: 'id'
      }
    }
  }, {
    sequelize,
    modelName: 'Nota',
    tableName: 'notas',
    timestamps: true,
    createdAt: 'data_criacao',
    updatedAt: 'updated_at'
  });

  return Nota;
};

File: backend-go/internal/models/nota.go
package models

import "time"

// Nota espelha a tabela `notas`. O model Sequelize original NÃO declarava
// `nova`/`destinatario`, mas o código das rotas (notas.js/notasController) os
// usa — presumimos colunas reais na tabela (ver spec §4.2/§6.11). Se a migration
// real não tiver essas colunas, remover os campos ou marcá-los com `gorm:"-"`.
//
// Timestamps: createdAt está mapeado para a coluna `data_criacao` (nome custom
// do Sequelize), updatedAt para `updated_at`.
type Nota struct {
	ID           uint    `gorm:"primaryKey" json:"id"`
	TenantID     *uint   `gorm:"column:tenant_id;index" json:"tenant_id,omitempty"`
	ClienteID    uint    `gorm:"column:cliente_id;not null;index" json:"cliente_id"`
	ProcessoID   *uint   `gorm:"column:processo_id" json:"processo_id"`
	Texto        string  `gorm:"column:texto;not null" json:"texto"`
	CriadoPorID  *uint   `gorm:"column:criado_por_id" json:"criado_por_id"`
	Nova         *bool   `gorm:"column:nova;default:true" json:"nova"`
	Destinatario *string `gorm:"column:destinatario" json:"destinatario"`

	DataCriacao time.Time `gorm:"column:data_criacao" json:"data_criacao"`
	UpdatedAt   time.Time `gorm:"column:updated_at" json:"updated_at"`

	// Associações (não persistidas)
	Cliente *Cliente `gorm:"foreignKey:ClienteID;references:ID" json:"cliente,omitempty"`
	Criador *User    `gorm:"foreignKey:CriadoPorID;references:ID" json:"criador,omitempty"`
}

func (Nota) TableName() string { return "notas" }
